target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -0,0 +1,750 @@
+	REG_SET(400000000, 0x0414c, 0x00000005),
+	REG_SET(425000000, 0x04164, 0x00000005),
+	REG_SET(450000000, 0x0417c, 0x00000005),
+	REG_SET(475000000, 0x04194, 0x00000005),
+	REG_SET(500000000, 0x041ac, 0x00000005),
+	REG_SET(525000000, 0x041c4, 0x00000005),
+	REG_SET(550000000, 0x0412c, 0x00000004),
+	REG_SET(575000000, 0x0413c, 0x00000004),
+	REG_SET(600000000, 0x0414c, 0x00000004),
+	REG_SET(625000000, 0x0415c, 0x00000004),
+	REG_SET(650000000, 0x0416c, 0x00000004),
+	REG_SET(675000000, 0x0417c, 0x00000004),
+	REG_SET(700000000, 0x0418c, 0x00000004),
+	REG_SET(725000000, 0x0419c, 0x00000004),
+	REG_SET(750000000, 0x041ac, 0x00000004),
+	REG_SET(775000000, 0x041bc, 0x00000004),
+	REG_SET(800000000, 0x041cc, 0x00000004),
+	REG_SET(825000000, 0x041dc, 0x00000004),
+	REG_SET(850000000, 0x041ec, 0x00000004)
+};
+
+static const struct rtcl_reg_set rtcl_839x_mem_reg_set[] = {
+	REG_SET(100000000, 0x041CC, 0x00000000),
+	REG_SET(125000000, 0x041AC, 0x00000007),
+	REG_SET(150000000, 0x0414C, 0x00000006),
+	REG_SET(175000000, 0x0418C, 0x00000006),
+	REG_SET(200000000, 0x041CC, 0x00000006),
+	REG_SET(225000000, 0x0417C, 0x00000005),
+	REG_SET(250000000, 0x041AC, 0x00000005),
+	REG_SET(275000000, 0x0412C, 0x00000004),
+	REG_SET(300000000, 0x0414C, 0x00000004),
+	REG_SET(325000000, 0x0416C, 0x00000004),
+	REG_SET(350000000, 0x0418C, 0x00000004),
+	REG_SET(375000000, 0x041AC, 0x00000004),
+	REG_SET(400000000, 0x041CC, 0x00000004)
+};
+
+static const struct rtcl_reg_set rtcl_839x_lxb_reg_set[] = {
+	REG_SET(50000000, 0x1414c, 0x00000003),
+	REG_SET(100000000, 0x0814c, 0x00000003),
+	REG_SET(150000000, 0x0414c, 0x00000003),
+	REG_SET(200000000, 0x0414c, 0x00000007)
+};
+
+struct rtcl_rtab_set {
+	int count;
+	const struct rtcl_reg_set *rset;
+};
+
+#define RTAB_SET(_rset)				\
+	{					\
+		.count = ARRAY_SIZE(_rset),	\
+		.rset = _rset,			\
+	}
+
+static const struct rtcl_rtab_set rtcl_rtab_set[SOC_COUNT][CLK_COUNT] = {
+	{
+		RTAB_SET(rtcl_838x_cpu_reg_set),
+		RTAB_SET(rtcl_838x_mem_reg_set),
+		RTAB_SET(rtcl_838x_lxb_reg_set)
+	}, {
+		RTAB_SET(rtcl_839x_cpu_reg_set),
+		RTAB_SET(rtcl_839x_mem_reg_set),
+		RTAB_SET(rtcl_839x_lxb_reg_set)
+	}
+};
+
+#define ROUND_SET(_min, _max, _step)	\
+	{				\
+		.min = _min,		\
+		.max = _max,		\
+		.step = _step,		\
+	}
+
+struct rtcl_round_set {
+	unsigned long min;
+	unsigned long max;
+	unsigned long step;
+};
+
+static const struct rtcl_round_set rtcl_round_set[SOC_COUNT][CLK_COUNT] = {
+	{
+		ROUND_SET(300000000, 625000000, 25000000),
+		ROUND_SET(200000000, 375000000, 25000000),
+		ROUND_SET(100000000, 200000000, 25000000)
+	}, {
+		ROUND_SET(400000000, 850000000, 25000000),
+		ROUND_SET(100000000, 400000000, 25000000),
+		ROUND_SET(50000000, 200000000, 50000000)
+	}
+};
+
+static const int rtcl_divn3[] = { 2, 3, 4, 6 };
+static const int rtcl_xdiv[] = { 2, 4, 2 };
+
+/*
+ * module data structures
+ */
+
+#define COMPAT_RTL838X "realtek,rtl8380-clock"
+#define COMPAT_RTL839X "realtek,rtl8390-clock"
```

Comment:
is this a thing now? :p probably not do this, so that 'grep' works better

Reply:
In my opinion using a constant 4 times thorughout the code justifies a define. But if you like.

Reply:
I think the whole idea is that the string IS the constant is kinda the idea behind compatible strings. I get what you're saying though; its just not commonly done, probably for `git grep` reasons.